services/shhext/chat/crypto/crypto.go

```diff
@@ -0,0 +1,92 @@
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"errors"
+)
+
+const (
+	aesNonceLength = 12
+)
+
+func containsOnlyZeros(data []byte) bool {
+	for _, b := range data {
+		if b != 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func validateDataIntegrity(k []byte, expectedSize int) bool {
+	if len(k) != expectedSize {
+		return false
+	}
+	if expectedSize > 3 && containsOnlyZeros(k) {
+		return false
+	}
+	return true
+}
+
+func generateSecureRandomData(length int) ([]byte, error) {
+	res := make([]byte, length)
+
+	_, err := rand.Read(res)
+	if err != nil {
+		return nil, err
+	} else if !validateDataIntegrity(res, length) {
+		return nil, errors.New("crypto/rand failed to generate secure random data")
+	}
+	return res, nil
+}
+
+func EncryptSymmetric(key []byte, plaintext []byte) ([]byte, error) {
```

Comment:
Not aware if we have any rule regarding this, but it's probably more useful to have the public functions appear first in a file.